### DISCUSSION TITLE
Host my website on my server

### DISCUSSION THREAD

**Discussion by purvi-vaishnani17** [remix-run/remix#2258] [Q&A]
Can you help me with how to host react remix website on my server?

**Comment by kiliman** [comment]
I would look at any tutorials on how to host a Node Express server. Once you have that, you're good.

**Reply by purvi-vaishnani17** [reply]
Actually, I have cPanel access and there is a create a "Node.js Application" so I have only the option. In that page, I just have to set the entry file and the node application will be running. But for the remix application, to serve, I need to run "remix-serve build", and in the Node.js application creator, the is no option available to run the command, it accepts only entry file (index.js / server.js - as we usually have). So, what things can I do?

**Reply by kiliman** [reply]
Ah, I see. So you'll have to build the Remix app, then upload the files to your host. Do they have FTP? You can use a GitHub action so that as soon as you push, it'll build you app and deploy it.

I'll see if I can come up with a deploy script for you.

**Reply by purvi-vaishnani17** [reply]
Moving the remix app build files to the hosting is not an issue. We have the FTP access, File Manager access in the cPanel and git actions as well. As I see, there is no index.html file in the build folder but index.js. In the cPanel, I can create a Nodejs App and it will be run on the Passenger. While creating the Node.js app, I tried to set the entry file like "build/index.js" but it didn't work. When I use the command "remix-serve build", then it works! But I have to serve the remix app without "remix-serve build" command. What I can do is set the "build/index.js" path when creating the node application and trying to serve the remix app through the nodejs app.
We don't have SSH root access, so we can't perform other action. Whatever we have to do is only available in the cPanel. It will be very helpful if there are another ways to serve the remix app on the cPanel.

**Reply by kiliman** [reply]
I have created a sample repo that shows you how to bundle your Remix app into a single file that can be deployed to your host. I also include a GitHub action that will bundle and FTP your code and assets. You just need to configure the FTP info.

https://github.com/kiliman/remix-ftp-deploy
